timestamp-impl/src/main/java/com/palantir/timestamp/LastReturnedTimestamp.java

```diff
@@ -23,10 +26,13 @@ public LastReturnedTimestamp(long timestamp) {
     }
 
     public synchronized void increaseToAtLeast(long newTimestamp) {
-        this.timestamp = Math.max(this.timestamp, newTimestamp);
+        long current = this.timestamp;
+        if (newTimestamp > current) {
+            this.timestamp = newTimestamp;
+        }
     }
 
-    public long get() {
+    public synchronized long get() {
```

Comment:
note to reviewers: previously there was a potential race condition as we were doing writes under `synchronized(this)` but reads were not -- this resolves that potential misuse.